Surface_mesh_topology/doc/Surface_mesh_topology/Surface_mesh_topology.txt

```diff
@@ -262,9 +285,51 @@ The canonical form of a curve is obtained by flattening its brackets, removing i
 
+\subsection SMTopology_Simplicity_Test Simplicity Test
+It can be shown that a closed curve is homotopic to a simple one if and only if its canonical form can be made intersection free via infinitesimal perturbations and some homotopy preserving operations. One can imagine each edge in the quadrangulation has width and each vertex in the quadrangulation has area so that paths visiting the same vertex/edge multiple times can avoid intersection within a vertex or an edge. See \cgalFigureRef{fig_perturbation_sample} for an example.
+\cgalFigureBegin{fig_perturbation_sample,perturbation_sample.svg}
+ Applying perturbation to reduce 2 intersections between red subpath and blue subpath.
+    \cgalFigureEnd
+
+Such perturbation can be expressed as orderings of edges from the canonical form which cross the same edge in the quadrangulation. The idea of the algorithm is to inductively build such ordering and try to avoid intersection as best as we can.
+
+\subsubsection SMTopology_Simplicity_Test_Primitive Detect Repetition
+It can be proven that if the canonical form can be expressed as concatenation of two or more copies of the same path, the path can never be made to be intersection free. So the first step of the algorithm is to detect repetition.
+
+Let \f$P\f$ be a path and let \f$+\f$ be the operator of concatenation. It can be shown that \f$P\f$ contains no repetition if and only if there are only 2 matchings of \f$P\f$ in \f$P+P\f$ (matching the first and the second copy). The algorithm resorts to the Knuth-Morris-Pratt algorithm to decide in linear time if the canonical form contains no repetition.
+
+\subsubsection SMTopology_Simplicity_Test_Switch Avoid Crossing by Switching
+Apart from applying perturbation, the algorithm also tries to avoid crossing using a homotopy-preserving operation known as <em>switch</em>. A switch is triggered when intersection could be avoided by turning a left L-shaped subpath into a right L-shaped subpath. See \cgalFigureRef{fig_switch_sample} for an example.
+\cgalFigureBegin{fig_switch_sample,switch_sample.svg}
+ Top-left, an intersection between red subpath and green subpath at the start of a left L-shaped red subpath. Top-right, switch the left L-shape to right L-shape to avoid the intersection. Bottom, no switch because no intersection can be avoided.
+    \cgalFigureEnd
+
+\subsubsection SMTopology_Simplicity_Test_Relative_Order Decide Relative Order
+Since the algorithm inductively builds an ordering, it has to determine relative order between the edge being processed and edges that has been ordered. There are three cases to consider.
```

Comment:
```suggestion
Since the algorithm inductively builds an ordering, it has to determine a relative order between the edge being processed and the edges that has been ordered. There are three cases to consider.
```